persist/persist_test.go

```diff
@@ -26,3 +28,123 @@ func TestIntegrationRandomSuffix(t *testing.T) {
 		file.Close()
 	}
 }
+
+// TestAbsolutePathSafeFile tests creating and committing safe files with
+// absolute paths.
+func TestAbsolutePathSafeFile(t *testing.T) {
+	tmpDir := build.TempDir(persistDir, "TestAbsolutePathSafeFile")
+	err := os.MkdirAll(tmpDir, 0700)
+	if err != nil {
+		t.Fatal(err)
+	}
+	absPath := filepath.Join(tmpDir, "test")
+
+	// Create safe file.
+	sf, err := NewSafeFile(absPath)
+	defer sf.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check that the name of the file is not equal to the final name of the
+	// file.
+	if sf.Name() == absPath {
+		t.Errorf("safeFile created with filename: %s has temporary filename that is equivalent to finalName: %s\n", absPath, sf.Name())
+	}
+
+	// Write random data to the file and commit.
+	data := make([]byte, 10)
+	rand.Read(data)
+	_, err = sf.Write(data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = sf.Commit()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check that the file exists and has same data that was written to it.
+	f, err := os.Open(absPath)
+	defer f.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+	dataRead := make([]byte, 11)
+	n, err := f.Read(dataRead)
```

Comment:
don't forget that you need `io.ReadAll` to guard against partial reads!

In this case it doesn't matter though, you should just use `ioutil.ReadFile`.